src/net.rs

```diff
@@ -98,17 +98,20 @@ impl Clone for Box<NetworkStream + Send> {
 }
 
 impl NetworkStream {
```

Comment:
I don't think these methods should be public. We should provide a safe, checked API.